src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Lower.java

```diff
@@ -4381,7 +4722,7 @@ public void visitSelect(JCFieldAccess tree) {
             TreeInfo.name(tree.selected) == names._super &&
             !types.isDirectSuperInterface(((JCFieldAccess)tree.selected).selected.type.tsym, currentClass);
         tree.selected = translate(tree.selected);
-        if (tree.name == names._class) {
+        if (tree.name == names._class && tree.selected.type.isPrimitiveOrVoid()) {
```

Comment:
not sure why this change was needed

Reply:
This is again lambda deduplication. The current code will take `String.class` and rewrite it from scratch, with a new symbol in it (which is what makes deduplication fail).